### DISCUSSION TITLE
A way to access static files from `node_modules` at runtime

### DISCUSSION THREAD

**Discussion by kuncevic** [vercel/community#5182] [Help]
### Summary

I am running a nodejs app build with `nestjs` that  uses `geoip-country` module. 

All works locally but when deploying to vercel I am seeing a runtime error `ENOENT: no such file or directory, open '/var/task/node_modules/geoip-country/data/geoip-country.dat' `. [geoip-country](https://github.com/sapics/geoip-country#custom-directory-for-database-files) supports custom data file config through env variables, however after setting that on the vercel project seems didn't help as seeing the same error.

Seems to me this is a common use-case, I am seeing a lot of issues like that round the web and I wonder what is the best way to solve that problem.

### Example

https://issues-beige.vercel.app/

### Steps to Reproduce

basically the problem stands around this lines:

```
import { lookup } from 'geoip-country';

lookup('192.168.0.1');
```

However here is a full reproduction `nestjs` source https://github.com/kuncevic/geoip-issue
The `release` folder contains the vercel prod build https://github.com/kuncevic/geoip-issue/tree/master/release that currently deployed into https://issues-beige.vercel.app

**Comment by kryptonic21** [comment]
hi seems to hot the same issues . do you solved it???

**Reply by kuncevic** [reply]
Because of that issue, for now, I've ditched `geoip-country` for simple http request to a public country ip lookup. However I do have a few more ideas to try and will update on that later once have time testing that out. I believe the real question here is figuring out the real 'node_modules' path....

**Comment by LoneRifle** [comment]
If the static files are known ahead of time, one workaround would be to deliberately access the files. It appears that you do not have to read them, just simply triggering an access through `fs.readdir()` or similar would suffice.

Example of this [here](https://github.com/demossg/mockpass-on-netlify/blob/5dcc13078fe583ffa3ed447ef9061c3e7947bd10/api/index.js#L1-L9).

**Comment by ebnazmul** [comment]
did you found any solution?

**Reply by nicky-ru** [reply]
I had the same error in a similar case, file tracing helped me as described [here](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions)
